src/renderer/opengl.rs

```diff
@@ -16,63 +17,202 @@ use glutin::{
+fn clamp_render_buffer_size(size: &PhysicalSize<u32>) -> PhysicalSize<u32> {
     PhysicalSize::new(
         size.width.clamp(1, MAX_RENDERBUFFER_SIZE),
         size.height.clamp(1, MAX_RENDERBUFFER_SIZE),
     )
 }
 
-impl Context {
-    pub fn window(&self) -> &Window {
-        &self.window
-    }
+fn get_proc_address(surface: &Surface<WindowSurface>, addr: &CStr) -> *const c_void {
+    GlDisplay::get_proc_address(&surface.display(), addr)
+}
+
+impl OpenGLSkiaRenderer {
+    pub fn new(window: WindowConfig, srgb: bool, vsync: bool) -> Self {
+        #[allow(irrefutable_let_patterns)] // This can only be something else than OpenGL on Windows
+        let config = if let WindowConfigType::OpenGL(config) = window.config {
+            config
+        } else {
+            panic!("Not an opengl window");
+        };
+        let window = window.window;
+        let gl_display = config.display();
+        let raw_window_handle = window.raw_window_handle();
+
+        let size = clamp_render_buffer_size(&window.inner_size());
+
+        let surface_attributes = SurfaceAttributesBuilder::<WindowSurface>::new()
+            .with_srgb(Some(srgb))
+            .build(
+                raw_window_handle,
+                NonZeroU32::new(size.width).unwrap(),
+                NonZeroU32::new(size.height).unwrap(),
+            );
+        let window_surface =
+            unsafe { gl_display.create_window_surface(&config, &surface_attributes) }
+                .expect("Failed to create Windows Surface");
+
+        let context_attributes = ContextAttributesBuilder::new()
+            .with_profile(GlProfile::Core)
+            .build(Some(raw_window_handle));
+        let context = unsafe { gl_display.create_context(&config, &context_attributes) }
+            .expect("Failed to create OpenGL context")
+            .make_current(&window_surface)
+            .unwrap();
+
+        // NOTE: We don't care if these fails, the driver can override the SwapInterval in any case, so it needs to work in all cases
+        // The OpenGL VSync is always disabled on Wayland and Windows, since they have their own
+        // implementation
+        let _ = if vsync && env::var("WAYLAND_DISPLAY").is_err() && OS != "windows" && OS != "macos"
+        {
+            window_surface
+                .set_swap_interval(&context, SwapInterval::Wait(NonZeroU32::new(1).unwrap()))
+        } else {
+            window_surface.set_swap_interval(&context, SwapInterval::DontWait)
+        };
 
-    pub fn resize(&self, width: NonZeroU32, height: NonZeroU32) {
-        GlSurface::resize(&self.surface, &self.context, width, height)
+        gl::load_with(|s| get_proc_address(&window_surface, CString::new(s).unwrap().as_c_str()));
+
+        let interface = skia_safe::gpu::gl::Interface::new_load_with(|name| {
+            if name == "eglGetCurrentDisplay" {
+                return std::ptr::null();
```

Comment:
Why is `eglGetCurrentDisplay` filtered out? Can't we let `get_proc_address` also find `eglGetCurrentDisplay`? (possibly this is a peculiarity of EGL, I've never used EGL)

Reply:
I have no idea, it was introduced here 
* https://github.com/neovide/neovide/pull/631

After some discussion here
* https://github.com/neovide/neovide/issues/491#issuecomment-830762584
* https://github.com/neovide/neovide/issues/661

I found no issues in the glutin repository.